cypress/integration/ui/view_dashboard/index.js

```diff
@@ -0,0 +1,93 @@
+import { Given, When, Then } from 'cypress-cucumber-preprocessor/steps'
+
+const antenatalCareDashboardRoute = '#/nghVC4wtyzi'
+const immunizationDashboardRoute = '#/TAMlzYkstb7'
+
+Given('I open the Antenatal Care dashboard', () => {
+    cy.visit('/')
+    cy.get('[data-test="dhis2-uicore-chip"]')
+        .contains('Antenatal Care')
+        .click()
+})
+
+Then('the Antenatal Care dashboard displays in view mode', () => {
+    cy.location().should(loc => {
+        expect(loc.hash).to.equal(antenatalCareDashboardRoute)
+    })
+
+    cy.get('[data-test="dhis2-dashboard-view-dashboard-title"]')
+        .should('be.visible')
+        .and('contain', 'Antenatal Care')
+    cy.get('.highcharts-background').should('exist')
+})
+
+When('I select the Immunization dashboard', () => {
+    cy.get('[data-test="dhis2-uicore-chip"]')
+        .contains('Immun')
+        .click()
+})
+
+Then('the Immunization dashboard displays in view mode', () => {
+    cy.location().should(loc => {
+        expect(loc.hash).to.equal(immunizationDashboardRoute)
+    })
+
+    cy.get('[data-test="dhis2-dashboard-view-dashboard-title"]')
+        .should('be.visible')
+        .and('contain', 'Immunization')
+    cy.get('.highcharts-background').should('exist')
+})
+
+When('I search for dashboards containing Immunization', () => {
+    cy.get('[data-test="dhis2-dashboard-search-dashboard-input"]').type('Immun')
+})
+
+Then('Immunization and Immunization data dashboards are choices', () => {
+    cy.get('[data-test="dhis2-uicore-chip"]')
```

Comment:
As mentioned above I would argue its better to append the id of the item that should be visible and target that specifically, which truly will make sure that 'Immunization and Immunization data dashboards are choices', instead of the current implementation which only checks that "two dashboards are choices"

Reply:
Also `.and('have.length', 2)` means adding another dashboard containing `Immun` will cause this to fail.